legacy-commands-v1/src/main/java/io/github/minecraftcursedlegacy/api/command/DispatcherRegistry.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2020 The Cursed Legacy Team.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.minecraftcursedlegacy.api.command;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Registry for for mapping commands to the relevant {@linkplain CommandDispatcher command dispatchers}.
+ * @since 1.1.0
+ */
+public class DispatcherRegistry {
```

Comment:
Perhaps move the dispatcher stuff into a sub-package?

Reply:
good idea

Reply:
I'm keeping CommandDispatchEvent with ChatEvent in the main package, however, unless you think otherwise is better

Reply:
actually, might be better to move it to the dispatcher package?

Reply:
I'm leaving that up to you

Reply:
I'd say keep it out.

Reply:
For now, since it is for command dispatchers, but not actually the command "Dispatcher" api provided, I'll keep it in root. Let me know if I should move it to the dispatcher package.

Reply:
I would keep it in root.